src/ui/public/query_bar/directive/query_popover.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { uiModules } from 'ui/modules';
+import { documentationLinks } from '../../documentation_links/documentation_links';
+import {
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiSwitch,
+  EuiLink,
+  EuiText,
+  EuiSpacer,
+} from '@elastic/eui';
+
+const luceneQuerySyntaxDocs = documentationLinks.query.luceneQuerySyntax;
+const kueryQuerySyntaxDocs = documentationLinks.query.kueryQuerySyntax;
+
+const module = uiModules.get('app/kibana', ['react']);
+module.directive('queryPopover', function (localStorage) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      language: '<',
+      onSelectLanguage: '&',
+    },
+    link: function ($scope, $element) {
+      $scope.isPopoverOpen = false;
+
+      function togglePopover() {
+        $scope.$evalAsync(() => {
+          $scope.isPopoverOpen = !$scope.isPopoverOpen;
+        });
+      }
+
+      function closePopover() {
+        $scope.$evalAsync(() => {
+          $scope.isPopoverOpen = false;
+        });
+      }
+
+      function onSwitchChange() {
```

Comment:
Wouldn't it be better to check the "checked" state here, instead of tracking the language state in this directive?

```js
function onSwitchChange({ target }) {
  const newLanguage = target.checked ? ? 'kuery' : 'lucene';
  ...
}
```

Reply:
That feels like it would break the EuiSwitch component's encapsulation. More specifically, it relies on the fact that EuiSwitch uses a checkbox type input element under the hood. It might be nice if EuiSwitch exposed a more simple callback that didn't leak implementation details, but as far as I can tell it doesn't. In `onSwitchChange` I'm just following the pattern shown in the EUI docs.

Reply:
> It might be nice if EuiSwitch exposed a more simple callback that didn't leak implementation details, but as far as I can tell it doesn't.
Agreed, that's an upstream issue.

> I'm just following the pattern shown in the EUI docs.
Fair enough, I hadn't looked in there.